### DISCUSSION TITLE
Q: terraform syntax highlighting and fuzzy search

### DISCUSSION THREAD

**Discussion by az-z** [neoclide/coc.nvim#4855] [Q&A]
Hello,
I hope you don't mind 2 questions:

1.  is there a way to have a syntax highlighting for the terraform code?
2.  is it possible to have a fuzzy search with the  terrafom ls server?

Thank you
AZ

**Comment by fannheyward** [comment]
If the terraform language server supports semanticTokens highlighting, coc.nvim can use this to highlight the code.

> a fuzzy search wiht terrafom ls

Can't understand, what's you want to search from the LS?

**Reply by az-z** [reply]
@fannheyward ,
 thank you for assisting with my questions.

For the fuzzy search question:
 
for example, type:
```
resource aws_
```

that will bring an alphabetical list of everything matching "aws_".  If i need to get an "aws_instance" resource, I need to type exactly that, e.g.: ``` aws_instance``.  

The use case here is to filter out the list based on what user types, e.g.:
```
resource  insta
```
as user types "insta" the list will offer "aws_instance" as a choice for auto completion ( among other options).  

Screenshots:
current functionality : ![image](https://github.com/neoclide/coc.nvim/assets/5010397/2e0e8dff-caf5-43af-b920-861598188ea2)
proposed functionality: 
![image](https://github.com/neoclide/coc.nvim/assets/5010397/42a346fc-4bed-4ca8-b330-6747ec9f6439)

**Reply by az-z** [reply]
Followup on the syntax highlighting question:

It looks like  they support the syntax tokens - https://github.com/hashicorp/terraform-ls/blob/main/docs/syntax-highlighting.md

**Reply by az-z** [reply]
In addition: is there a way to get rid of the pre-created "name"? As I type a resource name, e.g. "resource aws_instance" the "name" literal gets created ( see the snapshot above). That is not useful at all, since the name of the resource is pretty much unique. Currently, it is inserted for all resources. 
Since this is a very obvious situation, I'm certain that I just need to type some magic combination to change the "name" ...
